pkg/repository/repository.go

```diff
@@ -3,88 +3,156 @@
+	Host  string
+	Name  string
+	Owner string
 }
 
 // Parse extracts the repository information from the following
 // string formats: "OWNER/REPO", "HOST/OWNER/REPO", and a full URL.
 // If the format does not specify a host, use the config to determine a host.
 func Parse(s string) (Repository, error) {
+	var r Repository
+
 	if git.IsURL(s) {
 		u, err := git.ParseURL(s)
 		if err != nil {
-			return nil, err
+			return r, err
 		}
 
 		host, owner, name, err := git.RepoInfoFromURL(u)
 		if err != nil {
-			return nil, err
+			return r, err
 		}
 
-		return irepo.New(host, owner, name), nil
+		r.Host = host
+		r.Name = name
+		r.Owner = owner
+
+		return r, nil
 	}
 
 	parts := strings.SplitN(s, "/", 4)
 	for _, p := range parts {
 		if len(p) == 0 {
-			return nil, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
+			return r, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
 		}
 	}
 
 	switch len(parts) {
 	case 3:
-		return irepo.New(parts[0], parts[1], parts[2]), nil
+		r.Host = parts[0]
+		r.Owner = parts[1]
+		r.Name = parts[2]
+		return r, nil
 	case 2:
-		host, _ := auth.DefaultHost()
-		return irepo.New(host, parts[0], parts[1]), nil
+		r.Host, _ = auth.DefaultHost()
+		r.Owner = parts[0]
+		r.Name = parts[1]
+		return r, nil
 	default:
-		return nil, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
+		return r, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
 	}
 }
 
 // Parse extracts the repository information from the following
 // string formats: "OWNER/REPO", "HOST/OWNER/REPO", and a full URL.
 // If the format does not specify a host, use the host provided.
 func ParseWithHost(s, host string) (Repository, error) {
+	var r Repository
+
 	if git.IsURL(s) {
 		u, err := git.ParseURL(s)
 		if err != nil {
-			return nil, err
+			return r, err
 		}
 
 		host, owner, name, err := git.RepoInfoFromURL(u)
 		if err != nil {
-			return nil, err
+			return r, err
 		}
 
-		return irepo.New(host, owner, name), nil
+		r.Host = host
+		r.Owner = owner
+		r.Name = name
+
+		return r, nil
 	}
 
 	parts := strings.SplitN(s, "/", 4)
 	for _, p := range parts {
 		if len(p) == 0 {
-			return nil, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
+			return r, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
 		}
 	}
 
 	switch len(parts) {
 	case 3:
-		return irepo.New(parts[0], parts[1], parts[2]), nil
+		r.Host = parts[0]
+		r.Owner = parts[1]
+		r.Name = parts[2]
+		return r, nil
 	case 2:
-		return irepo.New(host, parts[0], parts[1]), nil
+		r.Host = host
+		r.Owner = parts[0]
+		r.Name = parts[1]
+		return r, nil
 	default:
-		return nil, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
+		return r, fmt.Errorf(`expected the "[HOST/]OWNER/REPO" format, got %q`, s)
+	}
+}
+
+// CurrentRepository uses git remotes to determine the GitHub repository
+// the current directory is tracking.
+func CurrentRepository() (Repository, error) {
```

Comment:
Perhaps this should be renamed to just `Current()` since it is now located in the `Repository` package?